django/db/migrations/autodetector.py

```diff
@@ -1304,19 +1302,17 @@ def suggest_name(cls, ops):
         represent. Names are not guaranteed to be unique, but put some effort
         into the fallback name to avoid VCS conflicts if possible.
         """
-        if len(ops) == 1:
-            if isinstance(ops[0], operations.CreateModel):
-                return ops[0].name_lower
-            elif isinstance(ops[0], operations.DeleteModel):
-                return "delete_%s" % ops[0].name_lower
-            elif isinstance(ops[0], operations.AddField):
-                return "%s_%s" % (ops[0].model_name_lower, ops[0].name_lower)
-            elif isinstance(ops[0], operations.RemoveField):
-                return "remove_%s_%s" % (ops[0].model_name_lower, ops[0].name_lower)
-        elif ops:
-            if all(isinstance(o, operations.CreateModel) for o in ops):
-                return "_".join(sorted(o.name_lower for o in ops))
-        return "auto_%s" % get_migration_name_timestamp()
+        if not ops:
+            return "empty"
+
+        suggested_names = (op.suggest_migration_name() for op in ops)
```

Comment:
`suggest_migration_name()` can return `None`, we should handle that here. If the first one returns `None` then the code below will fail (`None + str = 💥`).

This would be perfect for the walrus operator, but for now we could make this a generator and a list:

```python
raw_suggested_names = (op.suggest_migration_name() for op in ops)
suggested_names = [name for name in raw_suggested_names if name]

if not suggested_names:
   return '😭'

name = suggested_names[0]
... etc
```

Reply:
Yes good point. I've added that filtering. Rather than returning the crying emoji, or the previous string "empty", I've moved to using the old `auto_YYYYMMDD` as the fallback. This format remains in use for merge migrations too.